integration-tests/chain-signatures/tests/cases/nightly.rs

```diff
@@ -0,0 +1,57 @@
+use integration_tests_chain_signatures::MultichainConfig;
+use mpc_recovery_node::protocol::presignature::PresignatureConfig;
+use mpc_recovery_node::protocol::triple::TripleConfig;
+use test_log::test;
+
+use crate::actions::{self, wait_for};
+use crate::with_multichain_nodes;
+
+#[test(tokio::test)]
+#[ignore = "This is triggered by the nightly Github Actions pipeline"]
+async fn test_nightly_signature_production() -> anyhow::Result<()> {
+    const SIGNATURE_AMOUNT: usize = 10;
+    const NODES: usize = 8;
+    const THRESHOLD: usize = 4;
+    const MIN_TRIPLES: usize = 10;
+    const MAX_TRIPLES: usize = 2 * NODES * MIN_TRIPLES;
+
+    let triple_cfg = TripleConfig {
+        // This is the min triples required by each node.
+        min_triples: MIN_TRIPLES,
+        // This is the total amount of triples that will be generated by all nodes.
+        max_triples: MAX_TRIPLES,
+        // This is the amount each node can introduce a triple generation protocol into the system.
+        max_concurrent_introduction: 4,
+        // This is the maximum amount of triples that can be generated concurrently by the whole system.
+        max_concurrent_generation: 24,
+    };
+    let presig_cfg = PresignatureConfig {
+        // this is the min presignatures required by each node
+        min_presignatures: 10,
+        // This is the total amount of presignatures that will be generated by all nodes.
+        max_presignatures: 1000,
+    };
+
+    let config = MultichainConfig {
+        triple_cfg,
+        presig_cfg,
+        nodes: NODES,
+        threshold: THRESHOLD,
+    };
+
+    with_multichain_nodes(config, |ctx| {
+        Box::pin(async move {
+            let state_0 = wait_for::running_mpc(&ctx, Some(0)).await?;
+            assert_eq!(state_0.participants.len(), NODES);
+            wait_for::has_at_least_mine_triples(&ctx, triple_cfg.min_triples).await?;
+
+            for _ in 0..SIGNATURE_AMOUNT {
+                wait_for::has_at_least_mine_presignatures(&ctx, 3).await?;
+                actions::single_signature_production(&ctx, &state_0).await?;
```

Comment:
Maybe not await but collect no. of success and no. of failures (& reason)

Reply:
True, I'll add error logs to this